libs/libc/stdlib/lib_reallocarray.c

```diff
@@ -0,0 +1,65 @@
+/****************************************************************************
+ * libs/libc/stdlib/lib_reallocarray.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdlib.h>
+#include <unistd.h>
+
+#include "libc.h"
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: reallocarray
+ *
+ * Description:
+ *   The reallocarray function has the same functionality as realloc but
+ *   it fails safely if multiplication overflow occurs.
+ *
+ * Input Parameters:
+ *   ptr   - old memory to be reallocated and freed
+ *   nmemb - number of elements
+ *   size  - size of one element in bytes
+ *
+ * Returned Value:
+ *   Upon successful completion, the address of the re-allocated memory
+ *   is returned and previous pointer is freed. NULL is returned on error
+ *   with original block of memory left unchanged.
+ *
+ ****************************************************************************/
+
+FAR void *reallocarray(FAR void *ptr, size_t nmemb, size_t size)
+{
+  size_t res;
+  if (__builtin_mul_overflow(nmemb, size, &res))
```

Comment:
can not use __builtin_mul_overflow in the common code

Reply:
Updated to manual overflow computation, should be ok now.